src/main/java/org/wildfly/security/auth/server/event/RealmEvent.java

```diff
@@ -39,8 +41,9 @@ protected RealmEvent() {
      * @param <P> the visitor parameter type
      * @param <R> the visitor return type
      * @return the value returned from the visitor {@code handleXxx} method
+     * @throws RealmUnavailableException if the realm is not able to handle requests for any reason
      */
-    public <P, R> R accept(RealmEventVisitor<P, R> visitor, P param) {
+    public <P, R> R accept(RealmEventVisitor<P, R> visitor, P param) throws RealmUnavailableException {
```

Comment:
I don't like this API change because it implies that only realms will ever examine realm events, which, while possibly true, is unnecessarily constraining.  In my opinion, visitor implementations should either consume their exceptions, else a generic exception type should be given in the type declaration.  Thoughts?